src/main/java/org/radarbase/appserver/service/FcmDataMessageService.java

```diff
@@ -259,8 +259,18 @@ public void updateDeliveryStatus(String fcmMessageId, boolean isDelivered) {
 
     // TODO: Investigate if data messages/notifications can be marked in the state CANCELLED when deleted.
     @Transactional
-    public void deleteDataMessageByFcmMessageId(String fcmMessageId) {
-        this.dataMessageRepository.deleteByFcmMessageId(fcmMessageId);
+    public void deleteDataMessageById(long id) {
+        Optional<DataMessage> dataMessage =
+                this.dataMessageRepository.findById(id);
+
+        dataMessage.ifPresentOrElse(
```

Comment:
The JPA repository supports `exists` clause. SO you can add `existsById(Long id)` to the repository and directly call it here.